tasks/main.yml

```diff
@@ -16,9 +16,9 @@
     msg: 'Development environment is not allowed to run against test or production'
   tags: ['configure_puppet_client', 'testing' ]
   when:
-    - puppet_environment != "production"
-    - puppet_environment != "test"
-    - puppet_file_content.stdout is search('test') or puppet_file_content.stdout is search('production')
+    - puppet_environment is not search('^production(_[A-z,0-9]+)?$')
+    - puppet_environment is not search('^test(_[A-z,0-9]+)?$')
```

Comment:
With this code, I managed to configure test3.epouta.csc.fi to the CCCP_4429 branch, I'm not sure why. So this doesn't catch it correctly